src/NHibernate/Type/TimestampType.cs

```diff
@@ -1,142 +1,50 @@
-		protected virtual DateTimeKind Kind => DateTimeKind.Unspecified;
-
-		public TimestampType() : base(SqlTypeFactory.DateTime)
-		{
-		}
+		/// <inheritdoc />
+		public override string Name => "Timestamp";
 
-		public override object Get(DbDataReader rs, int index, ISessionImplementor session)
-		{
-			return DateTime.SpecifyKind(Convert.ToDateTime(rs[index]), Kind);
-		}
-
-		public override object Get(DbDataReader rs, string name, ISessionImplementor session)
-		{
-			return Get(rs, rs.GetOrdinal(name), session);
-		}
-
-		public override System.Type ReturnedClass
-		{
-			get { return typeof(DateTime); }
-		}
+		/// <inheritdoc />
+		protected override DateTime GetDateTimeToSet(object value, ISessionImplementor session) =>
+			(value as DateTime?) ?? Now;
 
 		/// <summary>
-		/// Sets the value of this Type in the DbCommand.
+		/// Retrieve the string representation of the timestamp object. This is in the following format:
+		/// <code>
+		/// 2011-01-27T14:50:59.6220000Z
+		/// </code>
 		/// </summary>
-		/// <param name="st">The DbCommand to add the Type's value to.</param>
-		/// <param name="value">The value of the Type.</param>
-		/// <param name="index">The index of the DbParameter in the DbCommand.</param>
-		/// <param name="session">The session for which the operation is done.</param>
-		/// <remarks>
-		/// No null values will be written to the DbCommand for this Type. 
-		/// </remarks>
-		public override void Set(DbCommand st, object value, int index, ISessionImplementor session)
-		{
-			st.Parameters[index].Value = (value is DateTime) ? value : DateTime.SpecifyKind(Now, Kind);
-		}
-
-		public override string Name
-		{
-			get { return "Timestamp"; }
-		}
-
-		public override string ToString(object val)
-		{
-			return ((DateTime) val).ToString("O");
-		}
-
-		public override object FromStringValue(string xml)
-		{
-			return DateTime.Parse(xml);
-		}
-
-		#region IVersionType Members
-
-		public object Next(object current, ISessionImplementor session)
-		{
-			return Seed(session);
-		}
-
-		public static DateTime Round(DateTime value, long resolution)
-		{
-			return value.AddTicks(-(value.Ticks % resolution));
-		}
-
-		public virtual object Seed(ISessionImplementor session)
-		{
-			if (session == null)
-			{
-				return Now;
-			}
-			return Round(Now, session.Factory.Dialect.TimestampResolutionInTicks);
-		}
-
-		public IComparer Comparator
-		{
-			get { return Comparer.DefaultInvariant; }
-		}
-
-		#endregion
-
-		public object StringToObject(string xml)
-		{
-			return FromStringValue(xml);
-		}
-
-		public override System.Type PrimitiveClass
-		{
-			get { return typeof(DateTime); }
-		}
-
-		public override object DefaultValue
-		{
-			get { return DateTime.MinValue; }
-		}
-
-		public override string ObjectToSQLString(object value, Dialect.Dialect dialect)
-		{
-			return '\'' + value.ToString() + '\'';
-		}
+		public override string ToString(object val) =>
+			((DateTime) val).ToString("O");
```

Comment:
Maybe the only significant change of `Timestamp` compared to the new `DateTimeType`.